app/src/main/java/org/covidwatch/android/ble/BluetoothManager.kt

```diff
@@ -1,127 +1,94 @@
 package org.covidwatch.android.ble
 
-import android.app.*
-import android.bluetooth.BluetoothAdapter
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
 import android.content.ComponentName
 import android.content.Context
 import android.content.Context.BIND_AUTO_CREATE
 import android.content.Intent
 import android.content.ServiceConnection
 import android.os.Build
 import android.os.IBinder
-import android.util.Log
 import androidx.core.app.NotificationCompat
 import androidx.core.content.ContextCompat.getSystemService
 import org.covidwatch.android.R
-import org.covidwatch.android.data.ContactEvent
-import org.covidwatch.android.data.ContactEventDAO
-import org.covidwatch.android.data.CovidWatchDatabase
 import org.covidwatch.android.presentation.MainActivity
-import org.tcncoalition.tcnclient.BluetoothService
-import org.tcncoalition.tcnclient.BluetoothService.LocalBinder
-import org.tcncoalition.tcnclient.cen.*
-import org.tcncoalition.tcnclient.toBytes
-import org.tcncoalition.tcnclient.toUUID
-import java.util.*
-import java.util.concurrent.TimeUnit
-
+import org.tcncoalition.tcnclient.bluetooth.BluetoothStateListener
+import org.tcncoalition.tcnclient.bluetooth.TcnBluetoothService
+import org.tcncoalition.tcnclient.bluetooth.TcnBluetoothService.LocalBinder
+import org.tcncoalition.tcnclient.bluetooth.TcnBluetoothServiceCallback
 
 interface BluetoothManager {
-    fun startAdvertiser(cen: Cen)
-    fun stopAdvertiser()
-
-    fun startService(cen: Cen)
+    fun startService()
     fun stopService()
```

Comment:
currently we either start or stop this service it on every launch depending whether app has location permission, which imo doesn't make sense. every start should have a stop, right?

Reply:
Valid questions.
You are almost right. I wanted to make this PR as isolated as possible because i saw many things in progress regarding permissions and where we start the service. Basically, this is a temp solution that binds to the service if somewhere in the code `preference_is_contact_event_logging_enabled` was set to `true`. So the bind part doesn't start service if it's already running. 

re always in background, we should have the way to stop it because ideally a user can opt out from the app activities. The app will be useless, but that's their right. 
Plus, it doesn't make sense to run the service when you are home right ? :)